routing/index_graph_starter.cpp

```diff
@@ -176,9 +255,45 @@ void IndexGraphStarter::FindPointsWithCommonFeature(Joint::Id jointId0, Joint::I
   CHECK(found, ("Can't find common feature for joints", jointId0, jointId1));
 }
 
+void IndexGraphStarter::AddAdditionalZeroEdges(FakeJoint const & fp, EndPointType endPointType)
+{
+  CHECK(!m_graph.IsFakeFeature(fp.m_point.GetFeatureId()), ());
+
+  bool const isJoint = fp.BelongsToGraph();
+
+  vector<DirectedEdge> edges;
+  if (isJoint)
+  {
+    m_graph.GetEdgeList(fp.m_jointId, endPointType == EndPointType::Start /* is outgoing */,
+                        true /* graphWithoutRestrictions */, edges);
+  }
+  else
+  {
+    m_graph.GetIntermediatePointEdges(fp.m_point, true /* graphWithoutRestrictions */, edges);
+  }
+
+  for (DirectedEdge const & edge : edges)
+  {
+    m_graph.ForEachEdgeMappingNode(edge, [&](DirectedEdge const & e){
+      if (edge == e)
+        return;
+
+      switch (endPointType) {
+      case EndPointType::Start:
+        AddZeroLengthOnewayFeature(isJoint ? edge.GetFrom() : edge.GetTo(),
```

Comment:
Yes. We need this zero length edges when start/finish of the route is set near edges appeared while applying a restrictions.

Let's consider the case when the start point of the route is added near a restriction. While restriction applying process edges near restrictions could be copied. 

There're to possibilities for start point:
- road point of the start is a joint: in this case we need to connect the start and all joints come from the start (because of edge copying while applying restrictions) with directed zero length edges (AddZeroLengthOnewayFeature(edge.GetFrom(), e.GetFrom()));

- road point of the start is not a joint. That means that the start point lies on an edge between two joints. In that case we can need to go along the edge to the next closest joint and connect the closest  joint with zero length directed edges with all joints come from the closest joint (AddZeroLengthOnewayFeature(edge.GetTo(), e.GetTo()));